frozen_mutation.hh

```diff
@@ -48,23 +48,26 @@ private:
     Consumer& _consumer;
     stop_iteration _stop_consuming = stop_iteration::no;
 
-    void flush_rows_and_tombstones(position_in_partition_view pos) {
+    stop_iteration flush_rows_and_tombstones(position_in_partition_view pos) {
         if (!_static_row.empty()) {
             auto row = std::move(_static_row.get_existing());
             _stop_consuming = _consumer.consume(static_row(std::move(row)));
+            if (_stop_consuming) {
+                return _stop_consuming;
+            }
         }
         if (_current_row) {
             auto row_entry = std::move(_current_row_entry);
             _current_row = nullptr;
-            if (!_stop_consuming) {
-                _stop_consuming = _consumer.consume(clustering_row(std::move(*row_entry)));
+            _stop_consuming = _consumer.consume(clustering_row(std::move(*row_entry)));
+            if (_stop_consuming) {
+                return _stop_consuming;
             }
         }
         _rt_gen.flush(pos, [this] (range_tombstone_change rt) {
-            if (!_stop_consuming) {
-                _stop_consuming = _consumer.consume(std::move(rt));
-            }
+            _stop_consuming = _consumer.consume(std::move(rt));
```

Comment:
After this change, _consumer.consume() may be called after consume() returned stop_iteration::yes. Isn't it a regression?